src/common-util/src/main/java/ee/ria/xroad/common/certificateprofile/impl/GenericCertificateProfileInfoProvider.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * The MIT License
+ * Copyright (c) 2019- Nordic Institute for Interoperability Solutions (NIIS)
+ * Copyright (c) 2018 Estonian Information System Authority (RIA),
+ * Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
+ * Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package ee.ria.xroad.common.certificateprofile.impl;
+
+import ee.ria.xroad.common.CodedException;
+import ee.ria.xroad.common.ErrorCodes;
+import ee.ria.xroad.common.certificateprofile.AuthCertificateProfileInfo;
+import ee.ria.xroad.common.certificateprofile.CertificateProfileInfoProvider;
+import ee.ria.xroad.common.certificateprofile.DnFieldDescription;
+import ee.ria.xroad.common.certificateprofile.SignCertificateProfileInfo;
+import ee.ria.xroad.common.identifier.ClientId;
+import ee.ria.xroad.common.util.CertUtils;
+
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x500.style.BCStyle;
+
+import javax.security.auth.x500.X500Principal;
+
+import java.security.cert.X509Certificate;
+
+/**
+ * Generic certificate profile
+ */
+public class GenericCertificateProfileInfoProvider
+        implements CertificateProfileInfoProvider {
+
+    @Override
+    public AuthCertificateProfileInfo getAuthCertProfile(AuthCertificateProfileInfo.Parameters params) {
+        return new GenericAuthCertificateProfileInfo(params);
+    }
+
+    @Override
+    public SignCertificateProfileInfo getSignCertProfile(SignCertificateProfileInfo.Parameters params) {
+        return new GenericSignCertificateProfileInfo(params);
+    }
+
+    /**
+     * Auth cert
+     *
+     * CN = serverCode
+     * C = country
+     * serialNumber = memberCode
+     * O = memberName
+     */
+    private static class GenericAuthCertificateProfileInfo extends AbstractCertificateProfileInfo
```

Comment:
Since the authentication certificate is just a TLS clientAuth + serverAuth certificate, it should include the DNS name if some commonly used CA is used to issue the certs. Also, country code and all the fields in the ServerId should be present in a generic profile. In that way, it's possible to identify the Security Server that the certificate was issued to.

In addition, CN is actually deprecated as a server DNS name and subjectAlternateName should be used instead.

For example, using the Icelandic [auth cert profile](https://github.com/nordic-institute/X-Road/blob/develop/src/common-util/src/main/java/ee/ria/xroad/common/certificateprofile/impl/IsAuthCertificateProfileInfo.java) and setting `setReadOnly(false)` for the country code would include all the necessary information.